src/main/java/org/openstreetmap/atlas/geography/atlas/delta/AtlasDeltaGenerator.java

```diff
@@ -1,28 +1,44 @@
 package org.openstreetmap.atlas.geography.atlas.delta;
 
+import java.io.IOException;
+import java.nio.file.Files;
 import java.nio.file.Path;
 import java.nio.file.Paths;
+import java.util.List;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.ForkJoinPool;
+import java.util.stream.Collectors;
 
+import org.apache.commons.io.FilenameUtils;
 import org.openstreetmap.atlas.geography.atlas.Atlas;
 import org.openstreetmap.atlas.geography.atlas.AtlasResourceLoader;
 import org.openstreetmap.atlas.streaming.resource.File;
 import org.openstreetmap.atlas.streaming.resource.FileSuffix;
 import org.openstreetmap.atlas.utilities.runtime.Command;
 import org.openstreetmap.atlas.utilities.runtime.CommandMap;
+import org.openstreetmap.atlas.utilities.time.Time;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
 /**
+ * This CLI will allow you to point to two atlases and generate a diff of the two. You can point to
+ * single atlas files or directories of atlas shards. If you are diffing shards for both your input
+ * and your output, we will process diffs for each shard individually in parallel.
+ *
  * @author matthieun
  * @author hallahan
  */
 public class AtlasDeltaGenerator extends Command
 {
+    private static final int THREADS = 8; // Tweak this if desired.
```

Comment:
I would make this configurable through a switch, and then just make this the default value. Why did you choose 8 by the way?

Reply:
Basically, that's what my laptop has, and the amount of memory used is not too bad. This is so when you're on a beefy server with many cores, you don't run out of memory. I can make this as a switch, good idea.